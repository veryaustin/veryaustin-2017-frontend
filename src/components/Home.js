import React from 'react';

const Home = () => {
  return (
    <div id="home">This is the Home Component</div>
  );
};

export default Home;
