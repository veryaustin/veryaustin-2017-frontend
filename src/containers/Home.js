import React, {Component, PropTypes} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import {fetchHome} from '../actions/homeActions';

class Home extends Component {
 // Ajax call to get API data before the component mounts
 componentWillMount() {
   this.props.fetchHome("home");
 }

  render() {
    return(
      <div id="home">This is the New Home Component</div>
    );
  }
}

// Bind actions and dispatch them
function mapDispatchToProps(dispatch) {
  return bindActionCreators({fetchHome}, dispatch);
}

//  Home Props Validation
Home.propTypes = {
  fetchHome: PropTypes.func,
  home: PropTypes.object
};

export default connect(null, mapDispatchToProps)(Home);