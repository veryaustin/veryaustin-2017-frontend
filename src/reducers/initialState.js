export default {
  pages: {
    home: {},
    work: {},
    about: {},
    contact: {}
  },
  fetching: {
    isLoading: true,
    error: null
  }
};
