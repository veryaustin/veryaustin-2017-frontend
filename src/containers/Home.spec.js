import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import initialState from '../reducers/initialState';
import Home from './Home';

describe('Home Component', () => {
  let middlewares, mockStore, store;
  beforeEach(() => {
    middlewares = [];
    mockStore = configureStore(middlewares);
    store = mockStore(initialState);
  });

  it('renders the home component', () => {
    const wrapper = shallow(<Home store={store} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});