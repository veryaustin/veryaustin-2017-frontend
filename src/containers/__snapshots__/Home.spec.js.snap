exports[`Home Component renders the home component 1`] = `
<Home
  fetchHome={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  } />
`;
